tests/rptest/tests/controller_log_limiting_test.py

```diff
@@ -283,19 +283,24 @@ def test_create_user_limit(self):
         wait_until(lambda: self.check_capacity_is_full(OPERATIONS_LIMIT),
                    timeout_sec=10,
                    backoff_sec=1)
+        success_count = 0
+        quota_error_count = 0
         for i in range(OPERATIONS_LIMIT * 2):
             try:
                 rpk.sasl_create_user(
                     f"testuser_{i}", "password",
                     self.redpanda.SUPERUSER_CREDENTIALS.algorithm)
             except RpkException as err:
-                if i >= OPERATIONS_LIMIT:
-                    assert 'Too many requests' in err.stderr
+                if 'Too many requests' in err.stderr:
+                    quota_error_count += 1
                 else:
                     raise err
             else:
-                if i >= OPERATIONS_LIMIT:
-                    assert 0, "Too Many Requests error must be raised"
+                success_count += 1
+
+        self.logger.info(f"{success_count=}, {quota_error_count=}")
+        assert success_count > 0
+        assert quota_error_count > 0
```

Comment:
So it asserts that it can make 1 to 3 successful requests out of 4. I understand 3 could sneak in if it took over 1 second to make them all. How can it be 1?

Reply:
> How can it be 1?

I guess it can't, but this is more of a sanity check. Updated the condition to be more stringent.